test/e2e/specs/nux.test.js

```diff
@@ -18,4 +39,145 @@ describe( 'NUX', () => {
 		const secondTipText = await page.$eval( '.nux-dot-tip', ( element ) => element.innerText );
 		expect( secondTipText ).toContain( 'You’ll find more settings for your page and blocks in the sidebar.' );
 	} );
+
+	it( 'should show "Got it" once all tips have been displayed', async () => {
+		for ( let i = 1; i < NUMBER_OF_TIPS; i++ ) {
+			await page.click( '.nux-dot-tip .components-button.is-link' );
+		}
+
+		// Make sure "Got it" button appears on the last tip.
+		const gotItButton = await page.$x( '//button[contains(text(), \'Got it\')]' );
+		expect( gotItButton ).toHaveLength( 1 );
+
+		// Click the "Got it button".
+		await page.click( '.nux-dot-tip .components-button.is-link' );
+
+		// Verify no more tips are visible on the page.
+		const nuxTipElements = await page.$$( '.nux-dot-tip' );
+		expect( nuxTipElements ).toHaveLength( 0 );
+
+		// Tips should not be marked as disabled, but when the user has seen all
+		// of the available tips, they will not appear.
+		const nuxTipsLocalStorage = await getTipsLocalStorage( page );
+		expect( nuxTipsLocalStorage.areTipsEnabled ).toEqual( true );
+		expect( Object.keys( nuxTipsLocalStorage.dismissedTips ) ).toHaveLength( NUMBER_OF_TIPS );
```

Comment:
I'm a little worried that parsing the JSON from local storage like this makes the test very coupled to how the feature is implemented.

We could avoid some of this by calling the selectors e.g. `wp.data.select( 'core/nux' ).areTipsEnabled()`.

But I guess it's no big deal to change the E2E test if the implementation is changed 😅

Reply:
That was the notion, that the test is including the offline functionality implemented in this reducer… it's a behaviour not found elsewhere in the code so I felt having a test here would at least make us aware of changing it.

But fair enough, I didn't think to check it with `wp.data`. I'll do that, we don't need to expose that implementation detail here. Come to think of it, resetting tips that way as well would be nice. I'll give it a go.